code/modules/xenoarchaeology/xenoartifact_console.dm

```diff
@@ -0,0 +1,248 @@
+/obj/item/circuitboard/computer/xenoartifact_console
+	name = "research and development listing console (Computer Board)"
+	icon_state = "science"
+	build_path = /obj/machinery/computer/xenoartifact_console
+
+/obj/item/circuitboard/machine/xenoartifact_inbox
+    name = "bluespace straythread pad (Machine Board)"
+    icon_state = "science"
+    build_path = /obj/machinery/xenoartifact_inbox
+    req_components = list(
+        /obj/item/stack/ore/bluespace_crystal = 1,
+        /obj/item/stock_parts/capacitor = 1,
+        /obj/item/stock_parts/manipulator = 1,
+        /obj/item/stack/cable_coil = 1)
+    def_components = list(/obj/item/stack/ore/bluespace_crystal = /obj/item/stack/ore/bluespace_crystal/artificial)
+
+/*
+    Current bandaid for shipping artifacts. Don't get too attached.
+    This code, and labeler, are probably the most problematic.
+*/
+
+/obj/machinery/computer/xenoartifact_console
+    name = "research and development listing console"
+    desc = "A science console used to source sellers, and buyers, for various blacklisted research objects."
+    icon_screen = "xenoartifact_console"
+    icon_keyboard = "rd_key"
+    circuit = /obj/item/circuitboard/computer/xenoartifact_console
+    
+    var/list/sellers[8]
+    var/list/buyers[8]
+    var/list/tab_index = list("Listings", "Export", "Linking")
+    var/current_tab = "Listings"
+    var/current_tab_info = "Here you can find listings for various research samples, usually fresh from the field. These samples aren't distrubuted by the Nanotrasen affiliated cargo system, so instead listing data is sourced from stray bluespace-threads."
+    var/obj/machinery/xenoartifact_inbox/linked_inbox
```

Comment:
Potential qdel if linked_inbox is deleted while being referenced here

Reply:
Could you elaborate on this, I don't quite understand what you're saying?